src/RecyclableMemoryStream.cs

```diff
@@ -615,6 +598,21 @@ async Task CopyToAsyncImpl(CancellationToken ct, long totalBytesToWrite)
                     return destination.WriteAsync(this.largeBuffer, (int)startPos, (int)count, cancellationToken);
                 }
             }
+            async Task CopyToAsyncImpl(Stream destination, long startPos, long count, CancellationToken cancellationToken)
```

Comment:
`static` local function could ensure nothing adds a closure capture in the future and expresses intent on not allocating closures

Reply:
It needs `this`. I'd rather not pass it explicitly :)

Reply:
Perhaps just make it a member function instead?